slick-postgres-driver/src/main/scala/org/hatdex/libs/dal/BaseSchemaMigrationImpl.scala

```diff
@@ -51,118 +51,76 @@ trait BaseSchemaMigrationImpl extends SchemaMigration {
-        Try(liquibase.dropAll())
-          .recover {
-            case e =>
-              liquibase.forceReleaseLocks()
-              logger.error(s"Error dropping all database information")
-              throw e
-          }
-        liquibase.forceReleaseLocks()
+        val dbConnection = db.createSession().conn
+        val liquibase = createLiquibase(dbConnection, "")
+        try liquibase.dropAll()
+        catch {
+          case NonFatal(th) =>
+            logger.error("Error dropping all database information", th)
+            throw th
+        } finally liquibase.forceReleaseLocks()
       }
     }
 
-    eventuallyEvolved.failed.foreach { e =>
-      logger.error(s"Error updating database: ${e.getMessage}")
-    }
-
-    eventuallyEvolved
-  }
-
   def rollback(changeLogFiles: Seq[String]): Future[Unit] = {
     logger.info(s"Rolling back schema migrations: ${changeLogFiles.mkString(", ")}")
     changeLogFiles.foldLeft(Future.successful(())) { (execution, evolution) => execution.flatMap { _ => rollbackDb(evolution) } }
   }
 
-  private def updateDb(diffFilePath: String, dbConnection: Connection): Future[Unit] = {
-    val eventuallyEvolved = Future {
-
-      logger.info(s"Liquibase running evolutions $diffFilePath on db: [${dbConnection.getMetaData.getURL}]")
-      val liquibase = blocking {
-        createLiquibase(dbConnection, diffFilePath)
-      }
+  private def updateDb(diffFilePath: String, dbConnection: Connection): Future[Unit] =
+    Future {
       blocking {
+        logger.info(s"Liquibase running evolutions $diffFilePath on db: [${dbConnection.getMetaData.getURL}]")
+        val liquibase = createLiquibase(dbConnection, diffFilePath)
         listChangesets(liquibase, new Contexts(changeContexts))
-        Try(liquibase.update(changeContexts))
-          .recover {
-            case e =>
-              liquibase.forceReleaseLocks()
-              logger.error(s"Error executing schema evolutions: ${e.getMessage}")
-              throw e
-          }
-        liquibase.forceReleaseLocks()
+        try liquibase.update(changeContexts)
+        catch {
+          case NonFatal(th) =>
+            logger.error(s"Error executing schema evolutions: ${th.getMessage}")
+            throw th
+        } finally liquibase.forceReleaseLocks()
       }
     }
 
-    eventuallyEvolved.failed.foreach { e =>
-      logger.error(s"Error updating database: ${e.getMessage}")
-    }
-
-    eventuallyEvolved
-  }
-
-  private def rollbackDb(diffFilePath: String): Future[Unit] = {
-    val eventuallyEvolved = Future {
-
-      val dbConnection = db.createSession().conn
-
-      logger.info(s"Liquibase rolling back evolutions $diffFilePath on db: [${dbConnection.getMetaData.getURL}]")
-      val liquibase = blocking {
-        createLiquibase(dbConnection, diffFilePath)
-      }
+  private def rollbackDb(diffFilePath: String): Future[Unit] =
+    Future {
       blocking {
+        val dbConnection = db.createSession().conn
```

Comment:
Non blocking: The ``` val dbConnection = db.createSession().conn ``` and ``` createLiquibase(dbConnection, diffFilePath)``` are repeated in many spots. would it be less DRY to move creating liquibase to a method ?

Reply:
not sure what you mean... `createLiquibase` is a method creating the `Liquibase` facade.
the SQL connection is created in each public method but that cannot be avoided.